library/ns1_record.py

```diff
@@ -450,17 +500,68 @@ def update(self, zone, record):
                     changed = True
                     args[key] = input_data
 
+        # create a new copy of the previously sanitized dict that will be updated with chaning args to support --diff
+        after_changes = record_data.copy()
```

Comment:
just to be safe, can we do a `deepcopy` here? the copy method is shallow. I don't think it's an issue, but seems easy to make sure it won't be

Reply:
Absolutely. Did some reading and understand the difference between the two copy types now. The change has been made and tested. It will be included in the next code push as `after_changes = copy.deepcopy(record_data)`.